stoix/systems/q_learning/ff_noisy_dqn.py

```diff
@@ -0,0 +1,569 @@
+import copy
+import time
+from typing import Any, Callable, Dict, Tuple
+
+import chex
+import flashbax as fbx
+import flax
+import hydra
+import jax
+import jax.numpy as jnp
+import optax
+from colorama import Fore, Style
+from flashbax.buffers.trajectory_buffer import BufferState
+from flax.core.frozen_dict import FrozenDict
+from jumanji.env import Environment
+from jumanji.types import TimeStep
+from omegaconf import DictConfig, OmegaConf
+from rich.pretty import pprint
+
+from stoix.base_types import (
+    ActorApply,
+    ExperimentOutput,
+    LearnerFn,
+    LogEnvState,
+    OffPolicyLearnerState,
+    OnlineAndTarget,
+)
+from stoix.evaluator import evaluator_setup, get_distribution_act_fn
+from stoix.networks.base import FeedForwardActor as Actor
+from stoix.systems.q_learning.dqn_types import Transition
+from stoix.utils import make_env as environments
+from stoix.utils.checkpointing import Checkpointer
+from stoix.utils.jax_utils import unreplicate_batch_dim, unreplicate_n_dims
+from stoix.utils.logger import LogEvent, StoixLogger
+from stoix.utils.loss import q_learning
+from stoix.utils.total_timestep_checker import check_total_timesteps
+from stoix.utils.training import make_learning_rate
+from stoix.wrappers.episode_metrics import get_final_step_metrics
+
+
+def get_warmup_fn(
+    env: Environment,
+    q_params: FrozenDict,
+    q_apply_fn: ActorApply,
+    buffer_add_fn: Callable,
+    config: DictConfig,
+) -> Callable:
+    def warmup(
+        env_states: LogEnvState, timesteps: TimeStep, buffer_states: BufferState, keys: chex.PRNGKey
+    ) -> Tuple[LogEnvState, TimeStep, BufferState, chex.PRNGKey]:
+        def _env_step(
+            carry: Tuple[LogEnvState, TimeStep, chex.PRNGKey], _: Any
+        ) -> Tuple[Tuple[LogEnvState, TimeStep, chex.PRNGKey], Transition]:
+            """Step the environment."""
+
+            env_state, last_timestep, key = carry
+            # SELECT ACTION
+            key, policy_key = jax.random.split(key)
```

Comment:
I would split this key here into three keys: key, policy_key, noise_key and use the noise_key as you've stated you would do in the PR description - rngs={"noise" : noise_key}